python/ml4ir/base/data/tfrecord_reader.py

```diff
@@ -358,4 +361,7 @@ def read(
             )
         )
 
+    # We apply prefetch as it improved train/test/validation throughput by 30% in some real model training.
+    dataset = dataset.prefetch(tf.data.experimental.AUTOTUNE)
```

Comment:
@mbrette I don't think we should have prefetch when we are not batching. I'm going to fix this in a follow up PR.

Reply:
@lastmansleeping Why ? (I guess this may not be useful, but does it hurt ?and if not, we should not make a special case)
Also when do we not batch ?